### PR TITLE
[PWA-1009] Localize Form Validators

### PR DESCRIPTION
**Pull request by anthoula**
## Description

Localize all strings within form validation of the application

## Related Issue

https://jira.corp.magento.com/browse/PWA-1009

### Verification Stakeholders

@dpatil-magento 
@sirugh 
### Verification Steps

1. Go to the forms in the app such as Create Account form.
2. Enter all supported invalid scenarios for the form fields.
3. Make sure form error validation messages are translated.

## Checklist

- [x] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/util/__tests__/formValidators.spec.js [hunk 1]
```diff
@@ -64,25 +64,25 @@ describe('isRequired', () => {
     test('it returns a string for an invalid string', () => {
```
Comment (sirugh) [inline packages/venia-ui/lib/util/__tests__/formValidators.spec.js:64]: Can you update all these test titles?

File: packages/venia-ui/lib/components/Field/message.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
+    const { formatMessage } = useIntl();
     const { asyncError, error } = fieldState;
     const errorMessage = error || asyncError;
 
     const classes = mergeClasses(defaultClasses, propClasses);
     const className = errorMessage ? classes.root_error : classes.root;
+    let translatedErrorMessage;
 
-    return <p className={className}>{errorMessage || children}</p>;
+    if (errorMessage) {
```
Comment (sirugh) [inline packages/venia-ui/lib/components/Field/message.js:18]: As written, this assumes `asyncError` is the same format, but I think @jimbo said that the fieldState no longer returns an `asyncError` prop.

Can you validate that? If it no longer has the prop, you can remove the supporting code here for `asyncError` and then your current conditional here works. However, If we have to continue supporting `asyncError` we should just use `if(error)` here and use its props within this conditional.

File: packages/venia-ui/i18n/en_US.json [hunk 3]
```diff
@@ -318,6 +317,17 @@
     "stockStatusMessage.message": "An item in your cart is currently out-of-stock and must be removed in order to Checkout.",
     "taxSummary.estimatedTax": "Estimated Tax",
     "taxSummary.tax": "Tax",
+    "validation.hasLengthAtLeast": "Must contain at least {value} character(s).",
+    "validation.hasLengthAtMost": "Must not exceed {value} character(s).",
+    "validation.hasLengthExactly": "Must contain exactly {value} character(s).",
+    "validation.invalidAbbrev": "State \"{value}\" is not an valid state abbreviation.",
```
Comment (sirugh) [inline packages/venia-ui/i18n/en_US.json:323]:
```suggestion
    "validation.invalidAbbreviation": "State \"{value}\" is not a valid state abbreviation.",
```

File: packages/venia-ui/lib/util/formValidators.js [hunk 4]
```diff
@@ -61,17 +83,31 @@ export const validateRegionCode = (value, values, countries) => {
...
+                'Country "US" does not contain any available regions.'
+        };
+        return invalidRegions;
     }
 
     const region = regions.find(({ code }) => code === value);
     if (!region) {
-        return `State "${value}" is not an valid state abbreviation.`;
+        const invalidAbbrev = {
+            id: 'validation.invalidAbbrev',
```
Comment (sirugh) [inline packages/venia-ui/lib/util/formValidators.js:106]:
```suggestion
            id: 'validation.invalidAbbreviation',
```

**Review by sirugh — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**